zcash_client_backend/src/data_api/chain.rs

```diff
@@ -175,7 +175,7 @@ where
     // comparing against the `validate_from` hash.
 
     block_source.with_blocks::<_, Infallible, Infallible>(
-        validate_from.map(|(h, _)| h),
+        validate_from.map(|(h, _)| h + 1),
```

Comment:
This change reflects a previous inconsistency between the `block_source` API and the implementation of this method. The height at which one starts validating (the first block that the block source inspects) must be one greater than the block for which the block hash is provided. However, it wasn't previously really a bug because the `block_source` implementation for zcash_client_sqlite _also_ had an off-by-one error that meant that this didn't cause problems.